src/lib/analysis-collector-base.ts

```diff
@@ -177,18 +177,29 @@ export abstract class AnalysisCollectorBase {
     this.logger.debug([command, ...args].join(' '));
     this.logger.debug(options);
 
-    return new Promise((resolve) => {
+    return new Promise((resolve, reject) => {
       const proc = cp.spawn(command, args, options);
       const chunks: string[] = [];
+      const errors: string[] = [];
 
       proc.stdout.on('data', (chunk) => {
         this.logger.debug(chunk);
         chunks.push(chunk);
       });
 
+      proc.stderr.on('data', (chunk) => {
+        this.logger.debug(chunk.toString());
+        errors.push(chunk.toString());
+      });
+
       proc.on('exit', (code) => {
         this.logger.debug(`${ command } exited with status code ${ code }`);
-        resolve(chunks.join('').trim());
+        if (code !== 0) {
+          reject(errors.join('').trim());
+        }
+        else {
+          resolve(chunks.join('').trim());
```

Comment:
Seems odd to join the array without anything.

Right now there will only ever be 1 `errors` or 1 `chunks`, but if we `.join('\n')` then the output will not be affected while providing newline separated errors or chunks if we end up with multiple in the future.

Reply:
This is now reflected in the most recent commit. 

I will create the ticket(s) for the other tools, unless you would like to see these changes implemented within this PR.

Reply:
New issues are fine.  :thumbsup: